http/vibe/http/common.d

```diff
@@ -336,13 +337,222 @@ class HTTPStatusException : Exception {
+	{
+		auto ret = new MultiPartBodyPart();
+		ret.headers["Content-Disposition"] = "form-data; name=\"" ~ field_name ~ "\"; filename=\"" ~ filename ~ "\"";
+		ret.headers["Content-Type"] = content_type;
+		ret.content = createMemoryStream((() @trusted => cast(ubyte[]) content)(), false);
+		return ret;
+	}
+
+	static MultiPartBodyPart singleFile(string field_name, string filename, string content_type, ubyte[] content)
+	{
+		auto ret = new MultiPartBodyPart();
+		ret.headers["Content-Disposition"] = "form-data; name=\"" ~ field_name ~ "\"; filename=\"" ~ filename ~ "\"";
+		ret.headers["Content-Transfer-Encoding"] = "binary";
+		ret.headers["Content-Type"] = content_type;
+		ret.content = createMemoryStream(content, false);
+		return ret;
+	}
+
+	static MultiPartBodyPart multipleFilesPart(Path file)
+	{
+		import vibe.inet.mimetypes : getMimeTypeForFile;
+		import vibe.core.file : openFile, FileMode;
+
+		auto ret = new MultiPartBodyPart();
+		ret.headers["Content-Disposition"] = "file; filename=\"" ~ file.head.name ~ "\"";
+		string type = getMimeTypeForFile(file.toString);
+		ret.headers["Content-Type"] = type;
+		if (!type.startsWith("text/"))
+			ret.headers["Content-Transfer-Encoding"] = "binary";
+		ret.content = cast(InputStream) openFile(file, FileMode.read);
+		return ret;
+	}
+
+	static MultiPartBodyPart multipleFilesPart(string filename, string content_type, InputStream stream)
+	{
+		auto ret = new MultiPartBodyPart();
+		ret.headers["Content-Disposition"] = "file; filename=\"" ~ filename ~ "\"";
+		ret.headers["Content-Type"] = content_type;
+		ret.content = stream;
+		return ret;
+	}
+
+	static MultiPartBodyPart multipleFilesPart(string filename, string content_type, string content)
+	{
+		auto ret = new MultiPartBodyPart();
+		ret.headers["Content-Disposition"] = "file; filename=\"" ~ filename ~ "\"";
+		ret.headers["Content-Type"] = content_type;
+		ret.content = createMemoryStream((() @trusted => cast(ubyte[]) content)(), false);
+		return ret;
+	}
+
+	static MultiPartBodyPart multipleFilesPart(string filename, string content_type, ubyte[] content)
+	{
+		auto ret = new MultiPartBodyPart();
+		ret.headers["Content-Disposition"] = "file; filename=\"" ~ filename ~ "\"";
+		ret.headers["Content-Transfer-Encoding"] = "binary";
+		ret.headers["Content-Type"] = content_type;
+		ret.content = createMemoryStream(content, false);
+		return ret;
+	}
+
+	static MultiPartBodyPart multipleFiles(string name, MultiPart multipart)
+	{
+		import vibe.stream.memory : createMemoryOutputStream;
+
+		auto ret = new MultiPartBodyPart();
+		string boundary = randomMultipartBoundary;
+		ret.headers["Content-Disposition"] = "form-data; name=\"" ~ name ~ "\"";
+		ret.headers["Content-Type"] = "multipart/mixed; boundary=\"" ~ boundary ~ "\"";
+		auto stream = createMemoryOutputStream();
+		multipart.write(boundary, stream);
+		ret.content = createMemoryStream(stream.data, false);
+		return ret;
+	}
```

Comment:
Looks like the last three overloads can just call the first one to avoid some code duplication. The change to be vibe-core compatible would otherwise add some more duplication for the required `InterfaceProxy` constructor call.

Reply:
I changed it to emplace now which seems to work and changed old code using direct checks on inheritance checks to now access the private typeinfo field of the proxy class because I don't want to create another PR on vibe-core which would take 3 years to merge...